Tests/UnitTests/Mocks/MockHTTPClient.swift

```diff
@@ -86,8 +86,13 @@ class MockHTTPClient: HTTPClient {
             .requestAddingNonceIfRequired(with: verificationMode)
             .withHardcodedNonce
 
+        let apiKeyStore = request.path.apiKeyStore
+        let apiKey = apiKeyStore.getAPIKey(from: self.apiKeys)
+
+        XCTAssertNotNil(apiKey, "Nil \(apiKeyStore) API key for path '\(request.path)'")
```

Comment:
Not a huge fan of this but I didn't want to overcomplicate the mock or duplicate the implementation of te actual `HTTPClient`. I believe this inconsistency shouldn't happen in unit tests, but I'm not 100% sure. I'm open to better ideas